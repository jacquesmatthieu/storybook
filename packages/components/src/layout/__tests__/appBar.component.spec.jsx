import React from 'react';
import { shallow } from 'enzyme';

import AppBar from '../appBar.component';

describe('AppBar', () => {
  let props;

  beforeEach(() => {
    props = {
      currentAppName: 'foo',
      apps: [
        { id: 'foo', name: 'Foo', route: 'foo', color: '#foo', logo: 'foo' },
        { id: 'bar', name: 'Bar', route: 'bar', color: '#bar', logo: 'bar' },
      ],
      user: {
        firstName: 'foo',
        lastName: 'bar',
        profilePictureUrl: 'baz',
      },
    };
  });

  const getWrapper = () => shallow(<AppBar {...props} />);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();
    const subWrapper = wrapper.find('appBarcomponent__AppSwitcher')
      .renderProp('children', { open: true, toggle: jest.fn() });

    // Then
    expect(wrapper).toMatchSnapshot();
    expect(subWrapper).toMatchSnapshot();
  });

  it('should render fixed version', () => {
    // Given
    props.fixed = true;

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
