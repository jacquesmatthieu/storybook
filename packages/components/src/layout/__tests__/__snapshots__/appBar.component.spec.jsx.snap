// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should render fixed version 1`] = `
<appBarcomponent__AppBarWrapper
  fixed={true}
>
  <appBarcomponent__AppSwitcher>
    <Component />
  </appBarcomponent__AppSwitcher>
  <appBarcomponent__AppBarLinks />
  <appBarcomponent__UserDropDownButton
    onClick={[Function]}
  >
    <appBarcomponent__Avatar
      src="baz"
    />
    <typographycomponent__Typography
      variant="header"
    >
      foo
       
      bar
    </typographycomponent__Typography>
    <appBarcomponent__UserDropdownIcon>
      <Icon
        direction="down"
        name="arrow"
        theme={
          Object {
            "palette": Object {},
          }
        }
      />
    </appBarcomponent__UserDropdownIcon>
  </appBarcomponent__UserDropDownButton>
</appBarcomponent__AppBarWrapper>
`;

exports[`AppBar should render properly 1`] = `
<appBarcomponent__AppBarWrapper
  fixed={false}
>
  <appBarcomponent__AppSwitcher>
    <Component />
  </appBarcomponent__AppSwitcher>
  <appBarcomponent__AppBarLinks />
  <appBarcomponent__UserDropDownButton
    onClick={[Function]}
  >
    <appBarcomponent__Avatar
      src="baz"
    />
    <typographycomponent__Typography
      variant="header"
    >
      foo
       
      bar
    </typographycomponent__Typography>
    <appBarcomponent__UserDropdownIcon>
      <Icon
        direction="down"
        name="arrow"
        theme={
          Object {
            "palette": Object {},
          }
        }
      />
    </appBarcomponent__UserDropdownIcon>
  </appBarcomponent__UserDropDownButton>
</appBarcomponent__AppBarWrapper>
`;

exports[`AppBar should render properly 2`] = `
<Fragment>
  <appBarcomponent__AppSwitcherButton
    data-test-id="appSwitcherButton"
    onClick={[MockFunction]}
  >
    <appBarcomponent__AppLogo
      alt="Foo logo"
      src="foo"
    />
    <typographycomponent__Typography
      variant="title"
    >
      Foo
    </typographycomponent__Typography>
    <appBarcomponent__DropDownIcon>
      <Icon
        direction="up"
        name="arrow"
        theme={
          Object {
            "palette": Object {},
          }
        }
      />
    </appBarcomponent__DropDownIcon>
  </appBarcomponent__AppSwitcherButton>
  <appBarcomponent__AppSwitcherMenu
    open={true}
  >
    <appBarcomponent__AppLink
      data-test-id="barAppLink"
      key="bar"
      to="bar"
    >
      <appBarcomponent__AppLinkLogo
        color="#bar"
        src="bar"
      />
      <appBarcomponent__AppName
        variant="title"
      >
        Bar
      </appBarcomponent__AppName>
      <appBarcomponent__DropDownIcon
        color="#bar"
      >
        <Icon
          direction="right"
          name="arrow"
          theme={
            Object {
              "palette": Object {},
            }
          }
        />
      </appBarcomponent__DropDownIcon>
    </appBarcomponent__AppLink>
  </appBarcomponent__AppSwitcherMenu>
</Fragment>
`;
